scripts/common.py

```diff
@@ -0,0 +1,143 @@
+"""
+Common Utilities for Collateral Management
+
+This module provides shared functionality for interacting with the Collateral smart contract.
+It includes utilities for:
+- Loading contract ABIs
+- Establishing Web3 connections
+- Managing accounts and transactions
+- Retrieving and processing blockchain events
+- Validating addresses and calculating checksums
+"""
+
+import os
+import pathlib
+import sys
+import json
+import hashlib
+import requests
+from web3 import Web3
+from eth_account import Account
+
+
+def load_contract_abi():
+    """Load the contract ABI from the artifacts file."""
+    try:
+        abi_file = pathlib.Path(__file__).parent.parent / "abi.json"
+        return json.loads(abi_file.read_text())
+    except FileNotFoundError:
+        print(
+            "Error: Contract ABI not found. Please run 'forge build' first.",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+
+def get_web3_connection():
+    """Get Web3 connection from RPC_URL environment variable."""
+    rpc_url = os.getenv("RPC_URL")
+    if not rpc_url:
+        print("Error: RPC_URL environment variable is not set", file=sys.stderr)
+        sys.exit(1)
+
+    w3 = Web3(Web3.HTTPProvider(rpc_url))
+    if not w3.is_connected():
+        print("Error: Failed to connect to the network", file=sys.stderr)
+        sys.exit(1)
+    return w3
+
+
+def get_account():
+    """Get account from PRIVATE_KEY environment variable."""
+    private_key = os.getenv("PRIVATE_KEY")
+    if not private_key:
+        print("Error: PRIVATE_KEY environment variable not set", file=sys.stderr)
+        sys.exit(1)
+    return Account.from_key(private_key)
+
+
+def validate_address_format(address):
+    """Validate if the given address is a valid Ethereum address."""
+    if not Web3.is_address(address):
+        print("Error: Invalid address", file=sys.stderr)
+        sys.exit(1)
+
+
+def build_and_send_transaction(
+    w3, function_call, account, gas_limit=100000, value=0
+):
+    """Build, sign and send a transaction.
+
+    Args:
+        w3: Web3 instance
+        function_call: Contract function call to execute
+        account: Account to send transaction from
+        gas_limit: Maximum gas to use for the transaction
+        value: Amount of ETH to send with the transaction (in Wei)
+    """
+    transaction = function_call.build_transaction(
+        {
+            "from": account.address,
+            "nonce": w3.eth.get_transaction_count(account.address),
+            "gas": gas_limit,
+            "gasPrice": w3.eth.gas_price,
+            "chainId": w3.eth.chain_id,
+            "value": value,
+        }
+    )
+
+    signed_txn = w3.eth.account.sign_transaction(transaction, account.key)
+    print(f"Signed transaction: {signed_txn}", file=sys.stderr)
```

Comment:
```suggestion ```
this print is not helpful for the user